opendevin/controller/agent_controller.py

```diff
@@ -225,6 +225,8 @@ async def set_agent_state_to(self, new_state: AgentState):
         ):
             # user intends to interrupt traffic control and let the task resume temporarily
             self.state.traffic_control_state = TrafficControlState.PAUSED
+            # User has chosen to deliberately continue - lets double the max iterations
+            self.state.max_iterations *= 2
```

Comment:
In my understanding, @li-boxuan 's suggestion is to expand max_iterations if iterations are hit, and (maybe) expand the budget if the budget is hit. The code is currently expanding iterations in both cases, I don't know about that, does it really make sense to do it when the user reaches the budget per task limit? They're separate limits afaik.
Maybe I misunderstand though, happy to be corrected.

Reply:
Thanks for the clarification - I'll look further

Reply:
How about instead of doubling, just add the `max_iterations` config value (or default) again?